src/h5web/visualizations/heatmap/ColorBar.tsx

```diff
@@ -0,0 +1,47 @@
+import React from 'react';
+import { range } from 'lodash-es';
+import { ScaleSequential, scaleLinear } from 'd3-scale';
+import { AxisRight } from '@vx/axis';
+import useMeasure from 'react-use-measure';
+import { adaptedNumTicks } from './utils';
+import styles from './HeatmapVis.module.css';
+
+interface Props {
+  className: string;
+  scale: ScaleSequential<string>;
+}
+
+function ColorBar(props: Props): JSX.Element {
+  const { className, scale } = props;
+  const interpolator = scale.interpolator();
+  const gradientColors = range(0, 1.1, 0.1)
+    .map(i => `${interpolator(i)}`)
+    .reduce((acc, val) => `${acc},${val}`);
+
+  const [gradientRef, { height: gradientHeight }] = useMeasure();
+
+  return (
+    <div className={className}>
+      <div className={styles.colorBar}>
+        <div
+          ref={gradientRef}
+          className={styles.gradient}
+          style={{
+            background: `linear-gradient(${gradientColors})`,
+          }}
+        />
+        <svg className={styles.colorBarAxis}>
+          <AxisRight
+            scale={scaleLinear()
+              .domain(scale.domain())
+              .range([gradientHeight, 0])}
```

Comment:
I don't think I can do that: `scale` is a `ScaleSequential` whereas I need a `LinearScale` here

Reply:
Ah right! So you don't actually need the original `scale` in this component. Perhaps it would make more sense to pass the interpolator and the domain separately?

Reply:
Maybe `generateColorScale` should become `findDomain` or something... In the end, you only need the `sequentialScale` inside `computeTextureData`.

Reply:
My point was that all I needed was inside `scale` so that it was easier to pass it around. But it might be clearer in the end to do as you told.